### PR TITLE
feat: gcp document store supports prefixes

### LINKED ISSUE #26136: [Doc Handling] GCP document store prefix
**Description**

GCP document store must support a config property to set a custom prefix inside the bucket, e.g.: `CAMUNDA_DOCUMENT_STORE_GCP_PREFIX=/temp`.

The prefix should be set to `temp`by default in SaaS, because at the moment all files we store have a default TTL of 30 days. In the future we will have to deal with files that only expire when their associated process instance is completed, and this will likely be implemented using lifecycle policies. Lifecycle policies are usually granular based on prefixes, so this will serve as a good basis for the future changes (no need to migrate files from the old policy)

### PR DESCRIPTION
**Pull request by chillleader**
## Description

This PR will add prefix support to the GCP document store implementation.

The default prefix value is "temp" - the idea is that for now, all documents in SaaS will be temporary, but later we will likely add a new policy for documents that expire when their process instance is finished. Adding a prefix early on will help avoid migrations in the future.

**⚠️ This PR is part of the PR train that needs to follow a specific order when merging ⚠️**

@mo-okocha @marcosgvieira Please merge the PRs in this specific order to avoid merge conflicts:

- https://github.com/camunda/camunda/pull/26336
- https://github.com/camunda/camunda/pull/26340
- https://github.com/camunda/camunda/pull/26351

The branches of each PR are pointed towards the branch of the previous PR - please wait for the previous PR to be merged and make sure the target branch of the second PR is changed to main. Only after that you can merge the next PR.

**This PR goes second.**

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #26136

### CODE REVIEW

**Comment by chillleader** [comment]
⚠️ This PR is part of the PR train that needs to follow a specific order when merging ⚠️

@mo-okocha @marcosgvieira Please merge the PRs in this specific order to avoid merge conflicts:

https://github.com/camunda/camunda/pull/26336
https://github.com/camunda/camunda/pull/26340

    https://github.com/camunda/camunda/pull/26351

The branches of each PR are pointed towards the branch of the previous PR - please wait for the previous PR to be merged and make sure the target branch of the second PR is changed to main. Only after that you can merge the next PR.

This PR goes second.

**Review by mo-okocha — APPROVED** [review]
Looks good

**Pull request merged**
